engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/resource/BracketNode.java

```diff
@@ -0,0 +1,73 @@
+package org.enso.interpreter.node.expression.builtin.resource;
+
+import com.oracle.truffle.api.dsl.CachedContext;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.Language;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.dsl.MonadicState;
+import org.enso.interpreter.node.callable.InvokeCallableNode;
+import org.enso.interpreter.node.callable.thunk.ThunkExecutorNode;
+import org.enso.interpreter.runtime.Context;
+import org.enso.interpreter.runtime.callable.argument.CallArgumentInfo;
+import org.enso.interpreter.runtime.callable.argument.Thunk;
+import org.enso.interpreter.runtime.callable.function.Function;
+import org.enso.interpreter.runtime.data.ManagedResource;
+import org.enso.interpreter.runtime.state.Stateful;
+
+@BuiltinMethod(
+    type = "Resource",
+    name = "bracket",
```

Comment:
Please re-export this from the standard library in `Base.Resource`

Reply:
I can only do that with shadow-definitions, as I don't want more indirection here, for performance.